src/metrics.py

```diff
@@ -2,26 +2,84 @@
 #  See LICENSE file for licensing details.
 
 """Models and functions for the metric events."""
+import json
+import logging
+import time
 
-from pydantic import BaseModel, conint
+import requests
+from pydantic import BaseModel, NonNegativeInt
+
+import promtail
+
+PROMTAIL_PUSH_API_URL = "http://localhost:3100/loki/api/v1/push"
+
+requests_session = None  # pylint: disable=invalid-name
```

Comment:
What is the reason for needing this please? It looks a bit like a global variable? Also, for any disabled linting messages, please include a reason why it is reasonable to disable the rule in this case (see standard here: https://github.com/canonical/is-charms-contributing-guide/blob/main/CONTRIBUTING.md#static-code-analysis)

Reply:
I have removed it.